### ISSUE TITLE
no-implicit-globals regression starting with v21.0.0

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#23]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/587

---

**What version of ESLint are you using?**
5.11.1

**What version of TypeScript are you using?**
3.1.1

**What version of `typescript-eslint-parser` are you using?**
Any version from 21.0.0 onwards

**What code were you trying to parse?**

```json
{
  "parser": "typescript-eslint-parser",
  "rules": {
    "func-style": ["error", "declaration"],
    "no-implicit-globals": "error"
  }
}
```

```js
function foo() {
  return "bar";
}

module.exports = foo;
```

**What did you expect to happen?**
No errors

**What happened?**
In `typescript-eslint-parser@20.1.1, code passes with no errors.

With `typescript-eslint-parser@21.0.0 onwards, the errors below occur:

```bash
1:1  error  Implicit global variable, assign as global property instead  no-implicit-globals
```
